errors.go

```diff
@@ -0,0 +1,50 @@
+package gophercloud
+
+type BaseError struct {
+	Function string
+	Argument string
+	Value    interface{}
+	Message  string
+}
+
+func (e *BaseError) Error() string {
+	return e.Message
+}
+
+type defaultError401 struct {
+	*UnexpectedResponseCodeError
+}
+type defaultError404 struct {
+	*UnexpectedResponseCodeError
+}
+type defaultError405 struct {
+	*UnexpectedResponseCodeError
+}
+
+func (e defaultError401) Error() string {
+	return "Authentication failed"
+}
+func (e defaultError404) Error() string {
+	return "Resource not found"
+}
+func (e defaultError405) Error() string {
+	return "Method not allowed"
+}
+
+// Error401er is the interface resource implement to override the error message
```

Comment:
Sounds like this comment is missing a word somewhere. "Error401er is the interface resource [..] implement..." Maybe "... is the interface resource _error types_ implement to override..."?

Ditto for the other two Error interfaces below.